src/ucp/core/ucp_am.c

```diff
@@ -944,10 +944,14 @@ UCS_PROFILE_FUNC(ucs_status_ptr_t, ucp_am_send_nbx,
     flags     = ucp_request_param_flags(param);
     attr_mask = param->op_attr_mask &
                 (UCP_OP_ATTR_FIELD_DATATYPE | UCP_OP_ATTR_FLAG_NO_IMM_CMPL);
-
+    if (flags & UCP_AM_SEND_REPLY) {
+        max_short = ucp_ep_config(ep)->am_u.max_eager_short - sizeof(ucs_ptr_map_key_t);
+    } else {
+        max_short = ucp_ep_config(ep)->am_u.max_eager_short;
+    }
```

Comment:
can avoid this extra branch and instead:
- update `ucp_am_try_send_short` which already contains branch for `UCP_AM_SEND_REPLY`
- update `ucp_am_send_req` invocation for reply protocol below in this func